crates/common/src/result.rs

```diff
@@ -163,13 +163,6 @@ impl From<core::convert::Infallible> for WasmError {
     }
 }
 
-#[cfg(not(target_arch = "wasm32"))]
```

Comment:
Moving this out of common and into host allows us to remove `wasmer` as a dependency of common, greatly reducing the number of ancestor crates that need a feature flag (~20 to ~3)